sky/provision/aws/config.py

```diff
@@ -191,10 +191,9 @@ def _get_role(role_name: str):
             for policy_arn in attach_policy_arns:
                 role.attach_policy(PolicyArn=policy_arn)
 
-            # SkyPilot: 'PassRole' is required by the head node to pass the role
-            # to the workers, so we can access S3 buckets on the workers.
-            # 'Resource' is to limit the role to only able to pass itself to the
-            # workers.
+            # SkyPilot: 'PassRole' is required by the controllers (jobs and
+            # services) created with `aws.remote_identity: SERVICE_ACCOUNT` to
```

Comment:
```suggestion
            # services) created with `remote_identity: SERVICE_ACCOUNT` to
```
this field seems available to gcp/k8s etc. too

Reply:
The `PassRole` is for AWS only, and will only be needed when `aws.remote_identity` is set to `SERVICE_ACCOUNT`.